### ISSUE TITLE
Unable to build mlx_lm from source

### ISSUE THREAD

**Issue by chimezie** [ml-explore/mlx-examples#477]
After successfully building mlx (on an Apple Mac Studio Z14K0007J M1 Ultra) from git main, I tried to building mlx_lm the way I usually do from my local copy of mlx-examples:

```bash
% pip install -U -e llms
Obtaining file:///Users/oori/medical_llm/mlx-examples/llms
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      Traceback (most recent call last):
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/oori/.local/venv/medical-finetune/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 132, in get_requires_for_build_editable
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 441, in get_requires_for_build_editable
          return self.get_requires_for_build_wheel(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-build-env-a3_2urc6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
        File "/Users/oori/medical_llm/mlx-examples/llms/mlx_lm/__init__.py", line 1, in <module>
          from .convert import convert
        File "/Users/oori/medical_llm/mlx-examples/llms/mlx_lm/convert.py", line 9, in <module>
          import mlx.core as mx
      ModuleNotFoundError: No module named 'mlx'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by awni** [comment]
Looks to me like it failed to build `mlx`. 

Can you try `python -c "import mlx.core as mx; print(mx.__version__)"`?

I don't know why the `mlx` package is not available. Maybe you can share more on how you built it?

**Comment by chimezie** [comment]
This is how I built mlx (after pulling changes from git/main):

```bash
% env MACOSX_DEPLOYMENT_TARGET=14.2.1 CC=gcc CXX=g++ SDKROOT=`xcrun --show-sdk-path` CMAKE_BUILD_PARALLEL_LEVEL="" pip install  .
[..]
  Building wheel for mlx (pyproject.toml) ... done
  Created wheel for mlx: filename=mlx-0.4.0.dev20240223+d729a199-cp311-cp311-macosx_14_0_universal2.whl size=33120117 sha256=e12f205546c293124e9b5b275a26c4fe40eaf81a5fbc62410b345d60a003c36a
  Stored in directory: /private/var/folders/4b/bht1mbdj6xd_1jmhss9cmpbw0000gn/T/pip-ephem-wheel-cache-625iz8uh/wheels/93/15/16/01908bf91d03701448751e2fe7ba9e19ef1ac36825365b7470
Successfully built mlx
Installing collected packages: mlx
  Attempting uninstall: mlx
    Found existing installation: mlx 0.4.0.dev20240223+d729a199
    Uninstalling mlx-0.4.0.dev20240223+d729a199:
      Successfully uninstalled mlx-0.4.0.dev20240223+d729a199
Successfully installed mlx-0.4.0.dev20240223+d729a199
```

After which I get the following:

```bash
% python -c "import mlx.core as mx; print(mx.__version__)"
0.4.0.dev20240223+d729a199
```

I tried building again and get the same error.  I'm even able to import mlx.core in the same way as the failing line:

```python
% ipython
Python 3.11.6 (v3.11.6:8b6ee5ba3b, Oct  2 2023, 11:18:21) [Clang 13.0.0 (clang-1300.0.29.30)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import mlx.core as mx

In [2]: 
```

Which makes me think it might be an import resolution issue related to the relative import: 

```python
 from .convert import convert
```

**Comment by chimezie** [comment]
I was able to build it after removing those unused imports from the top of __init__.py BTW:

```bash
% git diff
diff --git a/llms/mlx_lm/__init__.py b/llms/mlx_lm/__init__.py
index 8296042..311f216 100644
--- a/llms/mlx_lm/__init__.py
+++ b/llms/mlx_lm/__init__.py
@@ -1,4 +1 @@
-from .convert import convert
-from .utils import generate, load
-
 __version__ = "0.0.14"
```

**Comment by awni** [comment]
Yes, I see the problem. The fix is in #479

**Comment by 1060352728** [comment]
Try to remove the python variable from the local environment, and then recreate the virtual environment.
